### PR TITLE
Use slot in header usage instead of innerHTML

### PR DESCRIPTION
**Pull request by PatrykSzwer**
In my fork of this element I've changed implementation of header usage. Instead of using innerHTML I'm using slots. Thanks to that header can be declared like this:

```html
<div slot="header">
    <div class="someClass">something</div>
</div>
```
Which I think is more flexible and easier to use.

I've also removed `--layout-center-center` and added css for toogle arrow. For my needs It's just looking better that way.

### CODE REVIEW

File: paper-collapse-item.html [hunk 1]
```diff
@@ -71,6 +70,11 @@
 				@apply --paper-font-body1;
 				@apply --paper-collapse-item-content;
 			}
+			
+			.toogle {
+				position:absolute;
+				right:0px;
```
Comment (ankon) [inline paper-collapse-item.html]: Can you add a space after ':' to match the existing style?

**Review by ankon — CHANGES_REQUESTED** [review]
Would you mind fixing the typo 'toogle' -> 'toggle' in the CSS class while you're there?

File: paper-collapse-item.html (hunk 1 above)
Comment (PatrykSzwer) [reply paper-collapse-item.html]: Sure, It's fixed.

**Comment by PatrykSzwer** [comment]
I've updated readme file to present new usage of header.

**Comment by PatrykSzwer** [comment]
Is there anything more what I should change/fix/improve?

**Comment by warpech** [comment]
Ping. @ankon, @ronnyroeller can this be reconsidered?

**Comment by cwenzelg** [comment]
We would be also interested because this might impact SEO if the header is something like a h2.

**Comment by hyyan** [comment]
@warpech What is your use case for this? The header shoud be short , simple and none complicated.
Merging such a PR will break compatibility with older versions as we cannot keep using header property and slot together.

**Comment by cwenzelg** [comment]
for us, the use case is, that we use it for headers that we collapse on mobile and have open on a desktop for example. and therefore the heading will include html tags h2 for example. 
a second one is where we want a more complex header with a profile pic/icon for example (that would be another polymer element) and then open more details. All this would be easy to use, style and good for SEO. Also for additional accessibility.

hope that helped

**Comment by warpech** [comment]
For us the use case is to be able to provide any HTML for the header as demonstrated here in the usage of @PatrykSzwer's fork: 

https://github.com/Starcounter/Website/blob/c9fa87de8adec0b74bf99ed6c0570d5c941ca33e/src/WebsiteEditor/wwwroot/WebsiteEditor/views/BlendingPointsPage.html#L25-L33

Currently `paper-collapse-item` supports only text headers.

**Comment by ronnyroeller** [comment]
Guys - thanks for all your input. We don't have any of the stated requirements (SEO, complex headers) - so I don't have a strong opinion if this falls in the scope of the element or now.
In any case, we need to properly keep compatibility with older versions. Preferable, the feature gets implemented without breaking backwards compatibility. If everybody feels this is a really important feature and it can only be implemented as a breaking change, we could also consider creating a new major version.
What is your view?

**Comment by hyyan** [comment]
@ronnyroeller I think this is an important feature and it should be considered
but to keep breaking backward compatibility we can use both so we get the best of both worlds 

I remember paper-toast element provides something like that where you can set the `title` property 
and in the same time extend it with a slot.

Something like:

```html
<div class="html" inner-h-t-m-l="[[header]]"></div>
<slot name="header"></slot>
```

What do you think?

**Comment by PatrykSzwer** [comment]
@hyyan I've added backward compatibility with old solution. Please take a look.

**Comment by hyyan** [comment]
@PatrykSzwer Sounds OK for me, Thanks.
 @ronnyroeller Are you gonna merge? I believe the issue is resolved like this

**Pull request merged**